lib/chef-vault/item.rb

```diff
@@ -416,7 +416,7 @@ def node_exists?(nodename)
       # the node does not exist if a search for the node with that
       # name returns no results
       query = Chef::Search::Query.new
-      numresults = query.search(:node, "name:#{nodename}")[2]
+      numresults = query.search(:node, "name:#{nodename}", filter_result: { name: ["name"] }, rows: 0)[2]
```

Comment:
This will make SOLR return only the row count, which is the only thing we want (this is what the ```[2]``` selects). Way more efficient and faster.